database_integration/sqllite_db_interface.py

```diff
@@ -1,43 +1,72 @@
 import sqlite3
-import sys
 import os
-sys.path.append('..')
 from config import Config
-from typing import List, Dict 
+from typing import List, Dict,Set
+import pdb
 
 class SQLLiteDBInterface():
 
-    def __init__(self, db_name:str = None, table_name_caption_dict:Dict = None):
+    def __init__(self, db_name:str = None, table_name_schema_dict:Dict = None):
         # Connect to SQLite database (or create it if it doesn't exist)
         self.connection = sqlite3.connect(os.path.join(Config.sql_db_path, Config.sql_db_name if db_name is None else db_name))
-        self.cursor = connection.cursor()
+        self.cursor = self.connection.cursor()
 
-        if table_name_caption_dict None:
-            self.table_name_caption_dict = {Config.caption_table_name: Config.caption_table_schema, Config.processed_table_name : Config.processed_table_schema}
+        if table_name_schema_dict is None:
+            self.table_name_schema_dict = {Config.caption_table_name: [Config.caption_table_schema, Config.caption_table_pk]}
         else:
-            self.table_name_caption_dict = table_name_caption_dict
-        
+            self.table_name_schema_dict = table_name_schema_dict
         
         #create the table during the init
         self.create_table()
+        #otherwiese populate the table schema dictionary from before
+        if os.path.exists(os.path.join(Config.sql_db_path, Config.sql_db_name if db_name is None else db_name)):
```

Comment:
@Shreyas-S-Raman [CLEANING] move this to line 16 and make an if, elif, else block